inspirehep/modules/hal/tei.py

```diff
@@ -201,8 +183,8 @@ def _journal_data(pub_info):
 
 def _structure_data(struct):
     return {
-        'type': get_value(struct, "collections[1].primary", "").lower(),
-        # ^^ FIXME: This may not be one of the HAL accepted values:
+        'type': "institution",
+        # ^^ FIXME: Needs to choose from the HAL accepted values:
         # institution, department, laboratory or researchteam
```

Comment:
Here we miss to add the HAL ID of the institution, that is supposedly stored at the referenced recid.